### PR TITLE
Adds payment details entry UI for AU BECS Debit

### PR DESCRIPTION
**Pull request by csabol-stripe**
## Summary

Adds STPAUBECSDebitFormView that can be used to collect the required data to create an AU BECS Debit PaymentMethod.

## Motivation

Support for new payment method AU BECS Debit.

## Testing

Added examples to UI Examples and Non-Card Payment.
Added snapshot tests for new UI.
Added unit tests for view models and validators.
Tested on iOS 13, 12.2, and 10.2 simulators.

### CODE REVIEW

File: Example/UI Examples/AUBECSDebitFormViewController.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            ])
+        }
+
+        @objc func done() {
+            dismiss(animated: true, completion: nil)
+        }
+
+        override func viewDidAppear(_ animated: Bool) {
+            super.viewDidAppear(animated)
+//            becsFormView.becomeFirstResponder()
```
Comment (davidme-stripe) [inline Example/UI Examples/AUBECSDebitFormViewController.swift]: Should this be removed?

**Comment by davidme-stripe** [comment]
* `return` in the keyboard doesn't switch to the next field.
* When I finish typing the bank account number after typing an incorrect email address (with an accidental space in it), I get redirected to the email field. This seems weird?
* The selection/tap handlers are broken for all of these fields. When I try to move the cursor, it always highlights the entire field. I also can't seem to trigger the cut/copy/paste menu.

**Comment by csabol-stripe** [comment]
> Hitting return on the keyboard when typing doesn't switch to the next field.
Is this expected? Tabbing works

> When I finish typing the bank account number after typing an incorrect email address (with an accidental space in it), I get redirected to the email field. This seems weird?
This was intentional, copying the behavior of STPPaymentCardTextField moving between subfields, but I could see how it's more jarring in this layout. Do you think we should change to only ever auto-advance to next field if it's incomplete?

> The selection/tap handlers are broken for all of these fields. When I try to move the cursor, it always highlights the entire field. I also can't seem to trigger the cut/copy/paste menu.
Uh weird, I'll take a look. Thanks!

File: Example/UI Examples/AUBECSDebitFormViewController.swift (hunk 1 above)
Comment (csabol-stripe) [reply Example/UI Examples/AUBECSDebitFormViewController.swift]: oops thanks!

**Comment by davidme-stripe** [comment]
> > Hitting return on the keyboard when typing doesn't switch to the next field.
> 
> Is this expected? Tabbing works

It is: Try using it on a real iPhone without a physical keyboard attached, you'll see that hitting return is the usual UX for "next field".

> > When I finish typing the bank account number after typing an incorrect email address (with an accidental space in it), I get redirected to the email field. This seems weird?
> 
> This was intentional, copying the behavior of STPPaymentCardTextField moving between subfields, but I could see how it's more jarring in this layout. Do you think we should change to only ever auto-advance to next field if it's incomplete?
> 

Possibly!

> > The selection/tap handlers are broken for all of these fields. When I try to move the cursor, it always highlights the entire field. I also can't seem to trigger the cut/copy/paste menu.
> 
> Uh weird, I'll take a look. Thanks!

Looks like this happens for our other text fields, too, so maybe this could be fixed separately.

**Comment by csabol-stripe** [comment]
> Whoa, does this mean you managed to reproduce #1486?

Oh yeah, looks like it!

**Comment by csabol-stripe** [comment]
@davidme-stripe I'm gonna close this PR and open a new one because I think travis is broken on this one

**Pull request closed without merging**